lib/vfscore/stdio.c

```diff
@@ -211,15 +212,28 @@ static struct vfscore_file  stdio_file = {
 	.f_ep = UK_LIST_HEAD_INIT(stdio_file.f_ep)
 };
 
-void init_stdio(void)
+int init_stdio(void)
 {
-	int fd __maybe_unused;
+	int fd;
 
 	fd = vfscore_alloc_fd();
-	UK_ASSERT(fd == 0);
+	if (fd != 0) {
+		uk_pr_crit("failed to allocate fd for stdout\n");
+		return (fd < 0) ? fd : -EBADF;
+	}
 	vfscore_install_fd(0, &stdio_file);
-	if (uk_syscall_r_dup2(0, 1) != 1)
-		uk_pr_err("failed to dup to stdin\n");
-	if (uk_syscall_r_dup2(0, 2) != 2)
-		uk_pr_err("failed to dup to stderr\n");
+
+	fd = uk_syscall_r_dup2(0, 1);
+	if (fd != 1) {
+		uk_pr_crit("failed to dup to stdin\n");
```

Comment:
```suggestion
		uk_pr_crit("failed to dup to stdout\n");
```

Isn't fd==1 stdout or am I misreading the syscall?

Reply:
I'll update :+1: